tools/serverpod_cli/lib/src/migrations/generator.dart

```diff
@@ -122,19 +140,34 @@ class MigrationGenerator {
     required bool force,
     required int priority,
     bool write = true,
+    required GeneratorConfig config,
   }) async {
     var migrationRegistry = MigrationRegistry.load(
       migrationsProjectDirectory,
     );
 
+    var protocols =
+        await ProtocolHelper.loadProjectYamlProtocolsFromDisk(config);
+    var entityDefinitions = StatefulAnalyzer(protocols).validateAll();
+
+    var migrationVersions = await loadMigrationVersionsFromAllModules();
+
+    var installedModules = migrationVersions.values.map((m) {
+      return DatabaseMigrationVersion(
+        module: m.moduleName,
+        version: m.versionName,
+      );
+    }).toList();
+
     var srcDatabase = await _getSourceDatabaseDefinition(
       migrationRegistry.getLatest(),
       priority,
+      installedModules,
     );
 
-    var dstDatabase = await generateDatabaseDefinition(
-      directory: directory,
-      priority: priority,
+    var dstDatabase = createDatabaseDefinitionFromEntities(
+      entityDefinitions,
+      config,
```

Comment:
Suggestion: Still pass in priority here.

This PR shouldn't change the use of priority. That work should be done in a separate PR when the new SQL system is in place.